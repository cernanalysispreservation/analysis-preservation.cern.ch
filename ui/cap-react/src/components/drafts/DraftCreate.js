import React from "react";

import Layer from "grommet/components/Layer";

import CreateForm from "./components/CreateForm";

import PropTypes from "prop-types";

class DraftCreate extends React.Component {
  render() {
    return (
      <Layer overlayClose closer flush onClose={this.props.toggle}>
        <CreateForm />
      </Layer>
    );
  }
}

DraftCreate.propTypes = {
  contentTypes: PropTypes.object,
  toggle: PropTypes.func,
  createDraft: PropTypes.func,
  formData: PropTypes.object,
  metadata: PropTypes.object
};

export default DraftCreate;
