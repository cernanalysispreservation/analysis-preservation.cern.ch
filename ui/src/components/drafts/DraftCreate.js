import React from "react";

import { connect, Provider } from "react-redux";
import store from "../../store/configureStore";

import Anchor from "grommet/components/Anchor";
import Box from "grommet/components/Box";
import Heading from "grommet/components/Heading";
import Button from "grommet/components/Button";
import Layer from "grommet/components/Layer";
import Label from "grommet/components/Label";
import Paragraph from "grommet/components/Paragraph";
import CheckmarkIcon from "grommet/components/icons/base/Checkmark";
import CloseIcon from "grommet/components/icons/base/Close";

import FormField from "grommet/components/FormField";

import { TextInput } from "grommet";
import { createDraft } from "../../actions/draftItem";

class DraftCreate extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: "",
      type: null
    };
  }

  _selectType = type => {
    this.setState({ type });
  };

  _titleChange = event => {
    let title = event.target.value;
    this.setState({ title });
  };

  _createDraft = () => {
    let { title, type } = this.state;
    let data = {
      general_title: title
    };

    this.props.createDraft(data, { name: type });
  };

  render() {
    let { history, location } = this.props;

    return (
      <Layer overlayClose closer flush onClose={this.props.toggle}>
        <Provider store={store}>
          <Box pad="large" size={{ width: "xlarge" }} wrap={false} flex>
            <Box flex={false} size="xlarge">
              <Heading tag="h4">
                Select the content type you want to create, give a title to
                distinguish from other records and start preserving!
              </Heading>
              <Box pad={{ vertical: "small" }}>
                <FormField flex={false} label="General Title">
                  <Box pad={{ horizontal: "medium" }}>
                    <TextInput onDOMChange={this._titleChange} />
                  </Box>
                </FormField>
              </Box>
              <Box pad={{ vertical: "small" }}>
                <Box>
                  <Box>
                    <Box size={{ height: { max: "medium" } }}>
                      <Box
                        direction="row"
                        wrap
                        justify="between"
                        responsive={false}
                      >
                        {this.props.contentTypes &&
                          this.props.contentTypes.map(type => (
                            <Box
                              key={type.get("deposit_group")}
                              onClick={() =>
                                this._selectType(type.get("deposit_group"))
                              }
                              basis="1/2"
                              pad="small"
                              colorIndex="light-2"
                              separator="all"
                            >
                              <Box
                                flex
                                justify="between"
                                direction="row"
                                wrap={false}
                                style={{ overflow: "visible" }}
                              >
                                <Label size="small">{type.get("name")}</Label>
                                <Box flex={false} justify="center">
                                  {type.get("deposit_group") ==
                                  this.state.type ? (
                                    <CheckmarkIcon
                                      colorIndex="ok"
                                      size="xsmall"
                                    />
                                  ) : null}
                                </Box>
                              </Box>
                            </Box>
                          ))}
                      </Box>
                    </Box>
                  </Box>
                </Box>
              </Box>
            </Box>
            <Box align="end">
              <Button
                label="Start Preserving"
                onClick={this._createDraft}
                primary={true}
              />
            </Box>
          </Box>
        </Provider>
      </Layer>
    );
  }
}

function mapStateToProps(state) {
  return {
    id: state.draftItem.get("id"),
    errors: state.draftItem.get("errors"),
    contentTypes: state.auth.getIn(["currentUser", "depositGroups"])
  };
}

function mapDispatchToProps(dispatch) {
  return {
    createDraft: (data, type) => dispatch(createDraft(data, type))
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DraftCreate);
