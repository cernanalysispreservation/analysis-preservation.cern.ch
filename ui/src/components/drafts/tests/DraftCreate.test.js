import React from "react";
import { configure, shallow, render, mount } from "enzyme";
import { Map, List, fromJS } from "immutable";
import Adapter from "enzyme-adapter-react-16";

import DraftCreate from "../DraftCreate";
import { MemoryRouter as Router, Switch, Route } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
const mockStore = configureStore([]);

configure({ adapter: new Adapter() });

let wrapper;
let store;

store = mockStore({
  auth: Map({
    isLoggedIn: false,
    currentUser: Map({}),
    // token: localStorage.getItem("token"),
    error: null,
    loading: false,
    loadingInit: true,
    tokens: List(),
    integrations: Map({})
  }),
  draftItem: Map({
    errors: [],
    schemaErrors: [],
    actionsLayer: false,
    showPreviewer: false,
    filePreviewEditLayer: true,
    filePreviewEdit: {},

    workflows: fromJS([]),
    workflows_items: fromJS({}),
    bucket: Map({}),
    formData: null,
    // From backend: deposit resource
    access: null,
    can_admin: false,
    can_update: false,
    created: null,
    loading: false,
    created_by: null,
    experiment: null,
    files: Map({}),
    id: null,
    links: null,
    metadata: {},
    revision: null,
    schema: null,
    schemas: null,
    status: null,
    type: null,
    updated: null
  })
});

describe("Main Header Component", () => {
  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <DraftCreate />
      </Provider>
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("Should Match", () => {
    const shallowHeader = shallow(
      <Provider store={store}>
        <DraftCreate />
      </Provider>
    );

    expect(shallowHeader).toMatchSnapshot();
  });
});
