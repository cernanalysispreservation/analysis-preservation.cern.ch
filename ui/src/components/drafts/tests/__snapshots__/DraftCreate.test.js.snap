// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Header Component Should Match 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <Connect(DraftCreate) />
</Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Connect(DraftCreate) />,
      "value": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeState": Object {
          "auth": Immutable.Map {
            isLoggedIn: false,
            currentUser: Immutable.Map {
            },
            error: null,
            loading: false,
            loadingInit: true,
            tokens: Immutable.List [
            ],
            integrations: Immutable.Map {
            },
},
          "draftItem": Immutable.Map {
            can_admin: false,
            schema: null,
            access: null,
            created: null,
            bucket: Immutable.Map {
            },
            errors: Array [],
            schemaErrors: Array [],
            created_by: null,
            formData: null,
            metadata: Object {},
            schemas: null,
            status: null,
            workflows: Immutable.List [
            ],
            actionsLayer: false,
            workflows_items: Immutable.Map {
            },
            filePreviewEdit: Object {},
            files: Immutable.Map {
            },
            can_update: false,
            showPreviewer: false,
            links: null,
            type: null,
            filePreviewEditLayer: true,
            updated: null,
            id: null,
            revision: null,
            loading: false,
            experiment: null,
},
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": Object {},
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Connect(DraftCreate) />,
        "value": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "storeState": Object {
            "auth": Immutable.Map {
              isLoggedIn: false,
              currentUser: Immutable.Map {
              },
              error: null,
              loading: false,
              loadingInit: true,
              tokens: Immutable.List [
              ],
              integrations: Immutable.Map {
              },
},
            "draftItem": Immutable.Map {
              can_admin: false,
              schema: null,
              access: null,
              created: null,
              bucket: Immutable.Map {
              },
              errors: Array [],
              schemaErrors: Array [],
              created_by: null,
              formData: null,
              metadata: Object {},
              schemas: null,
              status: null,
              workflows: Immutable.List [
              ],
              actionsLayer: false,
              workflows_items: Immutable.Map {
              },
              filePreviewEdit: Object {},
              files: Immutable.Map {
              },
              can_update: false,
              showPreviewer: false,
              links: null,
              type: null,
              filePreviewEditLayer: true,
              updated: null,
              id: null,
              revision: null,
              loading: false,
              experiment: null,
},
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": Object {},
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
